dask/array/core.py

```diff
@@ -796,7 +805,9 @@ def __getitem__(self, index):
                 dt = np.dtype([(name, self._dtype[name]) for name in index])
             else:
                 dt = None
-            return elemwise(getarray, self, index, dtype=dt)
+            f = lambda x: getarray(x, index)
+            name = 'getitem-' + tokenize(self, index)
+            return elemwise(f, self, dtype=dt, name=name)
```

Comment:
Yep, it actually does work, as long as I don't coerce list arguments to dask arrays in elemwise.